modules/doc/content/getting_started/examples_and_tutorials/tutorial01_app_development/step07_parallel.md

```diff
@@ -1,6 +1,107 @@
 # Step 7: Execute in Parallel
 
-!alert construction
-The remainder of this tutorial is currently being developed. More content should be available soon. For now, refer back to the [examples_and_tutorials/index.md] page for other helpful training materials or check out the MOOSE [application_development/index.md] pages for more information.
+One of the major objectives of the MOOSE Framework is high performance. This step briefly introduces parallel processing with MOOSE. The basic commands used for running an application on multiple cores and/or multiple threads-per-core are demonstrated. A few basic tips on how to evaluate and improve performance are given.
+
+## MOOSE Multiprocessing
+
+At its core, MOOSE is designed to run in parallel by using the [Message Passing Interface](https://en.wikipedia.org/wiki/Message_Passing_Interface) protocol. [!ac](MPI) is a library of programming tools for accessing hardware and controlling how multiple CPUs exchange information while working simultaneously to run a single computer program. The general approach to solving a [!ac](FE) simulation in parallel is to partition the mesh and run an individual process that assembles and solves the system of equations for each of those mesh partitions. The hope is that the duration of solve procedures are decreased as the number of CPUs they engage is increased.
+
+### Model Setup id=model-setup
+
+The [Mesh] System in MOOSE provides several strategies for configuring a [!ac](FE) model to be solved in parallel. Most end-users won't have to alter the default settings. Even application developers need not worry about writing parallel code, since this is handled by the core systems of MOOSE, [libMesh], and [PETSc]. However, advanced users are likely to encounter situations in which the default parallelization techniques are not suitable for the problem they are solving. Such situations are beyond the scope of this tutorial and interested readers may refer to the following for more information:
+
+- [syntax/Mesh/Partitioner/index.md]
+- [syntax/Mesh/index.md#replicated-and-distributed-mesh]
+- [syntax/Mesh/splitting.md]
+- [source/partitioner/PetscExternalPartitioner.md]
+
+### Basic Commands id=commands
+
+A parallel execution is invoked at the command line along with the usual syntax for running an input file. There are two types of parallelism: [multiprocessing](https://en.wikipedia.org/wiki/Multiprocessing) and [multithreading](https://en.wikipedia.org/wiki/Thread_(computing%29). The former is enabled by the `mpiexec` command and the latter by the `--n-threads` option. For example, the following commands will run `simple_diffusion.i` on four CPUs, each with two threads:
```

Comment:
It could run on 4 CPUs with two threads but more likely it would run on 8 cores with a single thread?
I'd replace four CPUs by four processes

Reply:
But if you're assuming the user is running on a cluster with 4 machines assigned each a process instead of an 8-core machine, then CPUs is more meaningful / more clear to the tutorial reader.

Not sure what the most explicit way of phrasing this is. Maybe describe the assumed machine?